File: internal/aiagent/internal_secret_test.go
package aiagent

import "testing"

func TestDeriveInternalKeysStableVector(t *testing.T) {
	keys, err := DeriveInternalKeys("0123456789abcdef0123456789abcdef")
	if err != nil {
		t.Fatal(err)
	}
	if keys.ServiceToken != "rc9uB_qX7ORPNH5SB-_AhAPh3hgMj0qMkjfVkHqRDco" {
		t.Fatalf("unexpected service token: %s", keys.ServiceToken)
	}
	if keys.CallbackServiceToken == "" || keys.ActorSigningKey == "" {
		t.Fatalf("derived service identity keys are incomplete: %#v", keys)
	}
}

func TestDeriveInternalKeysRejectsShortSecret(t *testing.T) {
	if _, err := DeriveInternalKeys("too-short"); err == nil {
		t.Fatal("expected short secret to be rejected")
	}
}

File: luna-agent/src/internal-secret.test.ts
import { describe, expect, it } from "vitest"
import { deriveInternalKeys } from "./internal-secret.js"

describe("deriveInternalKeys", () => {
  it("matches the shared Go vector", () => {
    const keys = deriveInternalKeys("0123456789abcdef0123456789abcdef")
    expect(keys.serviceToken).toBe("rc9uB_qX7ORPNH5SB-_AhAPh3hgMj0qMkjfVkHqRDco")
    expect(keys.actorSigningKey).toHaveLength(43)
    expect(keys.callbackServiceToken).toHaveLength(43)
    expect(keys.toolArgumentsEncryptionKey).toHaveLength(32)
    expect(keys.callbackServiceToken).not.toBe(keys.serviceToken)
  })

  it("rejects short roots", () => {
    expect(() => deriveInternalKeys("too-short")).toThrow(/at least 32 bytes/)
  })
})
